packages/paste-website/src/components/ThemeSwitcher.tsx

```diff
@@ -1,13 +1,79 @@
 import * as React from 'react';
+import styled from '@emotion/styled';
+import {themeGet} from 'styled-system';
 import {ScreenReaderOnly} from '@twilio-paste/screen-reader-only';
+import {Box} from '@twilio-paste/box';
+import {Themes, ThemesType} from '../constants';
+import {useActiveSiteTheme} from '../context/ActiveSiteThemeContext';
 
 interface ThemeSwitcherProps {
   children?: React.ReactElement;
 }
+
+const StyledThemeSwitcherLabel = styled.label<{}>(props => ({
+  cursor: 'pointer',
+  display: 'inline-block',
+  padding: `${themeGet('space.space20')(props)} ${themeGet('space.space40')(props)}`,
+  '&:hover': {
+    textDecoration: 'underline',
+  },
+}));
+
+const StyledThemeSwitcherRadio = styled.input<{}>(props => ({
+  opacity: 0,
+  position: 'absolute',
+  [`&:focus + ${StyledThemeSwitcherLabel}`]: {
+    boxShadow: themeGet('shadows.shadowFocus')(props),
+    textDecoration: 'underline',
+  },
+  [`&:checked + ${StyledThemeSwitcherLabel}`]: {
+    backgroundColor: themeGet('backgroundColors.colorBackgroundPrimary')(props),
+    borderRadius: themeGet('radii.borderRadius10')(props),
+    color: themeGet('textColors.colorTextInverse')(props),
+  },
+}));
+
 export const ThemeSwitcher: React.FC<ThemeSwitcherProps> = () => {
+  const {theme, updateActiveSiteTheme} = useActiveSiteTheme();
+
+  const handleChange = (event: React.FormEvent<HTMLInputElement>): void => {
```

Comment:
Been doing some reading on Typescript and I wanted to offer an alternative typing approach similar to how we type components (`React.FC<>`):

```js
const handleChange: React.ChangeEventHandler<HTMLInputElement> = event => {
```



> The first method uses an inferred method signature (e: React.FormEvent<HTMLInputElement>): void and the second method enforces a type of the delegate provided by @types/react. So React.ChangeEventHandler<> is simply a "blessed" typing by @types/react, whereas you can think of the inferred method as more... artisanally hand-rolled. Either way it's a good pattern to know.

Source: https://github.com/typescript-cheatsheets/react-typescript-cheatsheet#forms-and-events

Reply:
I dig it